model/src/main/java/com/javahelp/model/token/Token.java

```diff
@@ -0,0 +1,123 @@
+package com.javahelp.model.token;
+
+import com.javahelp.model.user.User;
+import org.joda.time.DateTime;
+import java.security.SecureRandom;
+
+/**
+ * Representation of token entity
+ */
+
+public class Token {
+    /**
+     * Token requested by the user
+     */
+    private String token;
+
+    /**
+     * Unique id of the token.
+     */
+    private String id;
+
+    /**
+     * Expiry date of the token.
+     */
+    private DateTime expiry;
+
+    /**
+     * Issued date of the token.
+     */
+    private DateTime issued;
+
+    /**
+     * User who requested the token.
+     */
+    private User user;
+
+    /**
+     * This holds any additional information about the toekn.
+     */
+    private String tag;
+
+
+    /**
+     * Creates a new {@link Token}
+     * @param tag {@link String} of this {@link Token}
+     * @param user {@link User} who requested this {@link Token}
+     */
+    public Token(String tag, User user){
+        // Creating a token using functions from SecureRandom
+
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[20];
+        random.nextBytes(bytes);
+        this.token = bytes.toString();
```

Comment:
This may not do what you are thinking. You might want to read into the string representations of arrays in Java, and how they are represented by default. This string is not reflective of the contents of the array, and is also not really random.

Reply:
Yeah, it shows the memory address of the array I think. Is the usage of StringBuilder to append elements helpful here?

Reply:
Yuh exactly, and idk I mean we shouldn't be building the ID inside the token class, that's a job for the implementation of the gateway to the database